src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/LuceneDocumentConverterBase.cs

```diff
@@ -191,7 +193,21 @@ protected IEnumerable<AbstractField> GetRegularFields(IndexField field, object v
         private IEnumerable<AbstractField> GetComplexObjectFields(string path, Field.Store storage, BlittableJsonReaderObject val, Field.Index indexing, Field.TermVector termVector)
         {
             yield return GetOrCreateField(path + ConvertToJsonSuffix, TrueString, null, null, storage, Field.Index.NOT_ANALYZED_NO_NORMS, Field.TermVector.NO);
-            yield return GetOrCreateField(path, null, null, val.GetJsonReader(), storage, indexing, termVector);
+
+            StreamReader inMemoryComplexObject;
+            if (_complexInMemoryObjects.TryGetValue(path, out inMemoryComplexObject) == false)
```

Comment:
We need an in-memory buffer per field key because at the time of creating and adding Lucene document to an index its fields cannot share a single buffer. However that won't correctly work with arrays - I need to add support for and relevant test.